solidity/contracts/system/KeepFactorySelection.sol

```diff
@@ -34,29 +38,46 @@ library KeepFactorySelection {
 
         KeepFactorySelector factorySelector;
 
-        // Standard ECDSA keep factory: KEEP stake and ETH bond.
+        // Standard ECDSA keep vendor and factory: KEEP stake and ETH bond.
         // Guaranteed to be set for initialized factory.
+        IBondedECDSAKeepVendor keepStakeVendor;
         IBondedECDSAKeepFactory keepStakeFactory;
 
-        // Fully backed ECDSA keep factory: ETH stake and ETH bond.
-        IBondedECDSAKeepFactory ethStakeFactory;
+        // Fully backed ECDSA keep vendor and factory: ETH bond only.
+        IBondedECDSAKeepVendor fullyBackedVendor;
+        IBondedECDSAKeepFactory fullyBackedFactory;
+
+        // Lock for factories versions freeze. When set to true vendor won't be
+        // called to obtain a new factory address but a version of a factory
+        // from the moment of freezing will be used. Once locked the factory
+        // address won't be able to update anymore.
+        bool factoriesVersionsLock;
```

Comment:
Thoughts on naming it `factoriesVersionsLocked`?

`if (_self.factoriesVersionsLocked)` reads better than `if (_self.factoriesVersionsLock)`.

Reply:
88d7ca48e4f96d101b318379d39d6bea45a203c7